File: src/main/java/io/dustin/finstage/content/dto/ExternalContentListResponse.java
package io.dustin.finstage.content.dto;

import lombok.Getter;

import java.time.LocalDateTime;
import java.util.List;

@Getter
public class ExternalContentListResponse {
    private int total;
    private int page;
    private int size;
    private int totalPages;
    private boolean hasNext;
    private boolean hasPrev;
    private List<ExternalContentItem> items;


    @Getter
    public static class ExternalContentItem {
        private Long id;
        private String symbol;
        private String summary;
        private String url;
        private String title;
        private LocalDateTime date;

    }
}

File: src/main/kotlin/io/dustin/finstage/content/dto/ExternalContentListResponse.kt
package io.dustin.finstage.content.dto

import java.time.LocalDateTime

data class ExternalContentListResponse(
        val total: Int,
        val page: Int,
        val size: Int,
        val totalPages: Int,
        val hasNext: Boolean,
        val hasPrev: Boolean,
        val items: List<ExternalContentItem>
) {
    data class ExternalContentItem(
            val id: Long?,
            val symbol: String,
            val summary: String,
            val url: String,
            val title: String,
            val date: LocalDateTime
    )
}
